src/iterators.cpp

```diff
@@ -170,6 +128,42 @@ upstream_iterator& upstream_iterator::operator++()
 }
 
 
+// Instantiations
```

Comment:
Do you think it could be worthwhile to put this chunk in a vasculature_iterator file ?

Reply:
yeah, I think all the vasculature related things should be in vasculature; I'm not sure why they were in `types.h` in first place